testutil/server.go

```diff
@@ -98,6 +98,7 @@ type TestServerConfig struct {
 	VerifyOutgoing      bool                   `json:"verify_outgoing,omitempty"`
 	EnableScriptChecks  bool                   `json:"enable_script_checks,omitempty"`
 	Connect             map[string]interface{} `json:"connect,omitempty"`
+	EnableDebug         bool                   `json:"enable_debug,omitempty"`
```

Comment:
@i0rek Unfortunately it requires restarting the agent with that setting otherwise the application doesn't begin profiling. I'm going to add to this PR us always registering the handlers and just gating it on standard ACLs. 

@banks Given you brought this up in your review too, do you see any issues with doing an `operator:read` on the pprof endpoints (believe we'd need to duplicate and have handlers where we then call the pprof handler), and just always making them available without `enable_debug` being necessary at all? Or maybe just gate them on enable_debug public to all?

Reply:
@pearkes I meant that I would like to enable debug not only from a config file. If you want to turn it on real quick, an environment variable is easier. or commandline flag.

Reply:
https://golang.org/doc/diagnostics.html
> Can I profile my production services?

> Yes. It is safe to profile programs in production, but enabling some profiles (e.g. the CPU profile) adds cost. **You should expect to see performance downgrade**. The performance penalty can be estimated by measuring the overhead of the profiler before turning it on in production.

(my emphasis).

Note though that as far as I can tell that is only degrading performance _while the profile is running_. Actually registering the pprof handlers and including http/pprof doesn't do anything at all so I think it'd reasonable to always add the handlers and then gate access to them.

So I think `operator:read` is a good additional measure we could consider but we'd still need an explicit opt-in/out just have it gate the handler access instead of the initial registration?

I guess we'd need to make a little wrapper `http.HandlerFunc` that checked the config and optionally also checks for `operator:read` ACL in a header/query param and then if OK calls the pprof Handler directly.

We'd still need to default to off and optionally have an API that can flip the config so that we can enable it on the fly though - which kinda makes it a backdoor to getting pprof in any no-ACL cluster 🤔. 

We could _require_ that you have ACLs setup and operator:read OR config enabled I guess and then people with no ACLs (hopefully not too many real customers) still have the same experience as now (changing config on disk) but at least a HUP not a start/stop is sufficient. While people with ACLs can have the tool work automatically since they run this with an operator token and it's enabled by default then?

Reply:
Thanks for the thoughts @banks and @i0rek! 52d467b always registers the handlers and protects them with acls or enable_debug if ACLs are disabled.